hr_timesheet_hours_billed/models/sale_order_line.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2023-nowdays Cetmix OU (https://cetmix.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+from odoo.osv import expression
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    @api.depends(
+        "qty_delivered_method",
+        "analytic_line_ids.so_line",
+        "analytic_line_ids.unit_amount_billed",
+        "analytic_line_ids.product_uom_id",
+        "analytic_line_ids.approved",
+    )
+    def _compute_qty_delivered(self):
+        """This method compute the delivered quantity of the SO lines:
+        it covers the case provide by sale module, aka
+        expense/vendor bills (sum of unit_amount_billed of AAL), and manual case.
+        """
+
+        return super(SaleOrderLine, self)._compute_qty_delivered()
+
+    def _get_delivered_quantity_by_analytic(self, additional_domain):
+        """Compute and write the delivered quantity of current SO lines,
+        based on their related analytic lines.
+        :param additional_domain: domain to restrict AAL to include in computation
+        (required since timesheet is an AAL with a project ...)
+        """
```

Comment:
Doesn't seem to write. Plus, having a `get` method that writes can be misleading and sometime dangerous.
Regarding the format of docstring, I always recommend to follow https://peps.python.org/pep-0257/#multi-line-docstrings (which is kind of the same guideline we have for commit messages).

```suggestion
        """Retrieve delivered quantity by line.
        
        $longer-explanation-here-if-you-like

        :param additional_domain: domain to restrict AAL to include in computation
        (required since timesheet is an AAL with a project ...)
        """
```